device/common/include/traccc/finding/device/impl/propagate_to_next_surface.ipp

```diff
@@ -124,6 +142,13 @@ TRACCC_DEVICE inline void propagate_to_next_surface(
     } else {
         params_liveness[param_id] = 0u;
 
+        if (payload.debug) {
+            vecmem::device_atomic_ref<unsigned int>(
+                payload.debug->m_failure_count[static_cast<std::size_t>(
+                    propagate_to_next_surface_failure_mode::DETRAY_FAILURE)])
```

Comment:
Should this be tagged as a failure? I think this is where CKF propagation is terminated successfully

Reply:
Good point, I'll remove this state for now.